### PR TITLE
Feat/records filtering

### PR DESCRIPTION
**Pull request by StevenHee**
The search components have been changed, replacing the drop down menu with an icon toolbar. A filter icon has been added to toggle the visibility of the filter panel. An additional icon toggles the summary record columns between the initial fixed width view to responsive. This allows all the columns to be displayed on the screen without the need for horizontal scrolling. Permanently switching the table to responsive columns may impact users on lower resolution screens, hence the reason for allowing the user to toggle between the 2 states. The `Clear all` menu item has been replaced with an X icon within the search control to clear the search. This icon only appears when text is entered into the search input or the summary records are displaying search results.

The records filter panel displays a form containing the possible filters, which at this stage includes programme name only. To easily accommodate additional filters in future plus the possibility of similar filtering capabilities in connections, this filters form is dynamically generated and components have been added to display selection list and autocomplete controls. This approach also matches the current architecture of the records list components.

TICKET No. https://hee-tis.atlassian.net/browse/TIS21-2722

### CODE REVIEW

File: src/app/shared/form-controls/material-selection-list/material-selection-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Component, Input } from "@angular/core";
+import { FormGroup } from "@angular/forms";
+import { FormControlBase } from "../form-contol-base.model";
+
+@Component({
+  selector: "app-material-selection-list",
+  templateUrl: "./material-selection-list.component.html"
+})
+export class MaterialSelectionListComponent {
```
Comment (hihilary) [inline src/app/shared/form-controls/material-selection-list/material-selection-list.component.ts:9]: Is this new component designed for our future use?

File: src/app/shared/form-controls/material-selection-list/material-selection-list.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    initialValue: []
+  };
+  let data: any = {};
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [MaterialModule, FormsModule, ReactiveFormsModule],
+      declarations: [
+        MaterialSelectionListComponent,
+        MaterialSelectionListComponent
```
Comment (hihilary) [inline src/app/shared/form-controls/material-selection-list/material-selection-list.component.spec.ts:34]: `MaterialSelectionListComponent` is declared twice.

File: src/app/shared/form-controls/material-autocomplete/material-autocomplete.component.html [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      (click)="clearSelection()"
+    >
+      <mat-icon>close</mat-icon>
+    </button>
+
+    <mat-autocomplete
+      [panelWidth]="400"
+      #auto="matAutocomplete"
+      (optionSelected)="onSelected()"
+      [displayWith]="displayWith"
```
Comment (hihilary) [inline src/app/shared/form-controls/material-autocomplete/material-autocomplete.component.html:31]: The property and the method `displayWith` are not required.

**Review by CaiWillis — DISMISSED** [review]
Looks good and works locally 👍

File: src/app/shared/form-controls/material-selection-list/material-selection-list.component.ts (hunk 1 above)
Comment (StevenHee) [reply src/app/shared/form-controls/material-selection-list/material-selection-list.component.ts:9]: Yes. In light of forthcoming additional filters, I was keen to also test my approach using selection lists.

File: src/app/shared/form-controls/material-selection-list/material-selection-list.component.spec.ts (hunk 2 above)
Comment (StevenHee) [reply src/app/shared/form-controls/material-selection-list/material-selection-list.component.spec.ts]: Thanks. Fixed

File: src/app/shared/form-controls/material-autocomplete/material-autocomplete.component.html (hunk 3 above)
Comment (StevenHee) [reply src/app/shared/form-controls/material-autocomplete/material-autocomplete.component.html]: Ah yes. Thanks.

**Review by hihilary — APPROVED** [review]

**Pull request merged**